test/integration/integration.py

```diff
@@ -17,39 +17,56 @@
 
 class TestIntegration(unittest.TestCase):
     def test_integration(self):
-        rospy.logdebug("Integration Test Starting")
-
         rospy.init_node("integration_test")
 
-        rospy.loginfo("Integration Test Ready")
+        rospy.loginfo("Autonomy integration test starting...")
 
-        waypoint_in_world = SE3(position=np.array([-5.5, -5.5, 0.0]))
         waypoints = [
             (
                 Waypoint(tag_id=0, type=WaypointType(val=WaypointType.NO_SEARCH)),
-                waypoint_in_world,
+                SE3(position=np.array([4, 4, 0])),
+            ),
+            (
+                Waypoint(tag_id=0, type=WaypointType(val=WaypointType.POST)),
+                SE3(position=np.array([-2, -2, 0])),
             ),
+            (
+                Waypoint(tag_id=1, type=WaypointType(val=WaypointType.POST)),
+                SE3(position=np.array([11, -10, 0])),
+            )
         ]
-
-        publish_waypoints([convert_waypoint_to_gps(waypoint) for waypoint in waypoints])
+        publish_waypoints(list(map(convert_waypoint_to_gps, waypoints)))
 
         tf_buffer = tf2_ros.Buffer()
         tf2_ros.TransformListener(tf_buffer)
 
-        has_reached_target = False
-
-        while not rospy.is_shutdown():
-            try:
-                rover_in_world = SE3.from_tf_tree(tf_buffer, parent_frame="map", child_frame="base_link")
-                distance_to_target = waypoint_in_world.pos_distance_to(rover_in_world)
-                rospy.logdebug(distance_to_target)
-                if distance_to_target < COMPLETION_TOLERANCE:
-                    has_reached_target = True
-                if has_reached_target:
-                    rospy.signal_shutdown("Finished test")
-                    break
-            except tf2_ros.TransformException as exception:
-                pass
+        # def led_callback(msg: LED):
+        #     rospy.loginfo(f"LED message received: {msg}")
+        #
+        # rospy.Subscriber("/led", LED, led_callback)
+
+        rate = rospy.Rate(20)
+        for waypoint in waypoints:
+            rospy.loginfo(f"Moving to waypoint: {waypoint}")
+
+            def distance_to_current_waypoint():
```

Comment:
this is just checking the distance to the current waypoint but if we are driving to posts we would want the distance to the post right?

Reply:
Yes... that is a great point. I will ensure that the sim publishes all the urdfs to the tf tree, then we can check that.

Reply:
I think I fixed it with the latest updates